.github/workflows/smoke_test.yml

```diff
@@ -18,11 +20,30 @@ jobs:
 
     runs-on: ubuntu-20.04
 
+    # Our rspec is deeply coupled with a database. I had a go at decoupling it for smoke tests, and gave up.
```

Comment:
My guess is because `.rspec` loads everything
```
--require rails_helper
--require spec_helper
--require payment_calculator
```
it should really be only loading `spec_helper` then you'd probably be free to do what you want

Reply:
This might be one reason, but there is more. We clean database after each run for one
```
RSpec.configure do |rspec|
  rspec.before(:suite) { DatabaseCleaner.clean_with :truncation }
end
```
We check migration status
```
# Checks for pending migrations and applies them before tests are run.
# If you are not using ActiveRecord, you can remove these lines.
begin
  ActiveRecord::Migration.maintain_test_schema!
rescue ActiveRecord::PendingMigrationError => e
  puts e.to_s.strip
  exit 1
end
```
Aaand possibly do more. 

I think trying to cut all of them out of our rspec is, well, much more effort than giving the flow a dummy database, as much as it hurts my heart :(